assets/prototype/domain/eventEditor/ticketsList/ticketPriceCalculator/hooks/useOnSubmitPrices.js

```diff
@@ -14,41 +14,79 @@ const PRICE_INPUT_FIELDS = [
 	'wpUser',
 ];
 
+const parseBooleanField = (value) => {
+	value = typeof value === 'string' ? value.toLowerCase().trim() : value;
+	switch (value) {
+		case 'true':
+		case 'yes':
+		case '1':
+			return true;
+		case 'false':
+		case 'no':
+		case '0':
+			return false;
+		default:
+			return Boolean(value);
+	}
+};
+
 const useOnSubmitPrices = (existingPrices) => {
 	const { createEntity, updateEntity, deleteEntity } = useEntityMutator('Price');
 	const { updateEntity: updateTicket } = useEntityMutator('Ticket');
 	const existingPriceIds = existingPrices.map(({ id }) => id);
 
-	return (formData) => {
+	// Async to make sure that prices are handled before updating the ticket.
+	return async (formData) => {
 		console.log('%c Ticket Price Calculator Form Submit', 'color:YellowGreen;', formData);
 		const { ticket, prices = [] } = formData;
 
-		// Array containing the updated prices
-		const updatedPrices = [];
-
-		prices.forEach((price) => {
-			if (price.id === 'NEW_PRICE') {
-				return;
-			}
-			const { id, ...input } = pick(PRICE_INPUT_FIELDS, price);
-			// if it's a newly added price
-			if (!id) {
-				createEntity({ ...input, ticketId: ticket.id });
-			} else {
-				updateEntity({ id, ...input });
-			}
-			updatedPrices.push(id);
-		});
+		const updatedPriceIds = [];
+		const createdPriceIds = [];
+
+		// make sure to complete all price operations before updating the ticket
+		await Promise.all(
+			// covert the price operations into promises
+			prices.map((price) => {
+				if (price.id === 'NEW_PRICE') {
+					return Promise.resolve(price);
+				}
+				const { id, ...input } = pick(PRICE_INPUT_FIELDS, price);
+				// if it's a newly added price
+				if (!id) {
+					return new Promise((resolve, onError) => {
+						const onCompleted = ({ createPrice: { price } }) => {
+							createdPriceIds.push(price.id);
+							resolve(price);
+						};
+						createEntity({ ...input, ticketId: ticket.id }, { onCompleted, onError });
```

Comment:
looks like the field normalization I added got lost 😢

Reply:
🤦‍♂